crowbar_framework/app/models/ipmi_service.rb

```diff
@@ -48,9 +48,10 @@ def transition(inst, name, state)
     #
     if state == "discovering"
       @logger.debug("IPMI transition: discovering state for #{name}")
-      db = Proposal.where(barclamp: "ipmi", name: inst).first
       role = RoleObject.find_role_by_name "ipmi-config-#{inst}"
-      result = add_role_to_instance_and_node("ipmi", inst, name, db, role, "ipmi-discover")
+      Proposal.where(barclamp: "ipmi", name: inst).first do |db|
+        result = add_role_to_instance_and_node("ipmi", inst, name, db, role, "ipmi-discover")
```

Comment:
Lint/UselessAssignment: Useless assignment to variable - `result`. (https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars)